src/include/txn_inline.h

```diff
@@ -1352,14 +1360,39 @@ __wt_txn_modify_check(
     if (!rollback && upd == NULL) {
         __wt_read_cell_time_window(cbt, &tw, &tw_found);
         if (tw_found) {
-            if (WT_TIME_WINDOW_HAS_STOP(&tw))
+            if (WT_TIME_WINDOW_HAS_STOP(&tw)) {
                 rollback = !__wt_txn_tw_stop_visible(session, &tw);
-            else
+                if (rollback)
+                    __wt_verbose_debug(session, WT_VERB_TRANSACTION,
+                      "Conflict with update %" PRIu64 " at stop timestamp: %s", tw.stop_txn,
+                      __wt_timestamp_to_string(tw.stop_ts, ts_string));
+            } else {
                 rollback = !__wt_txn_tw_start_visible(session, &tw);
+                if (rollback)
+                    __wt_verbose_debug(session, WT_VERB_TRANSACTION,
+                      "Conflict with update %" PRIu64 " at start timestamp: %s", tw.start_txn,
+                      __wt_timestamp_to_string(tw.start_ts, ts_string));
+            }
         }
     }
 
     if (rollback) {
+        /* Dump information about the txn snapshot. */
+        if (WT_VERBOSE_LEVEL_ISSET(session, WT_VERB_TRANSACTION, WT_VERBOSE_DEBUG)) {
+            WT_ERR(__wt_scr_alloc(session, 1024, &buf));
+            WT_ERR(__wt_buf_fmt(session, buf,
+              "snapshot_min=%" PRIu64 ", snapshot_max=%" PRIu64 ", snapshot_count=%" PRIu32,
+              txn->snap_min, txn->snap_max, txn->snapshot_count));
+            if (txn->snapshot_count > 0) {
+                WT_ERR(__wt_buf_catfmt(session, buf, ", snapshots=["));
+                for (snap_count = 0; snap_count < txn->snapshot_count - 1; ++snap_count)
+                    WT_ERR(
+                      __wt_buf_catfmt(session, buf, "%" PRIu64 ",", txn->snapshot[snap_count]));
+                WT_ERR(__wt_buf_catfmt(session, buf, "%" PRIu64 "]", txn->snapshot[snap_count]));
+                __wt_verbose_debug(session, WT_VERB_TRANSACTION, "%s", (const char *)buf->data);
```

Comment:
The `__wt_verbose_debug` must be outside of the `if (txn->snapshot_count > 0)`. It is possible that there may not be any running transactions when the snapshot is acquired.

Reply:
You are right, my bad!